### PR TITLE
feat(cli): Unify CLI auth methods

### PR DESCRIPTION
**Pull request by DanielHougaard**
# Description 📣

We've previously only had support for token-based auth in the CLI for a specific set of operations _(mostly operations we deemed as important for CI/CD workflows)_. As Infisical grows, so does our users use-cases, and we need to allow all our CLI operations to be used via. token-based auth. This PR addresses the lack of a unified auth system in the CLI (#1802). 

## Type ✨

- [x] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation

---

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

### CODE REVIEW

File: cli/packages/cmd/folder.go [hunk 1]
```diff
@@ -95,19 +102,31 @@ var createCmd = &cobra.Command{
...
+
 		params := models.CreateFolderParameters{
 			FolderName:  folderName,
-			WorkspaceId: workspaceFile.WorkspaceId,
 			Environment: environmentName,
 			FolderPath:  folderPath,
+			WorkspaceId: projectId,
+		}
+
+		if token != nil && (token.Type == util.SERVICE_TOKEN_IDENTIFIER || token.Type == util.UNIVERSAL_AUTH_TOKEN_IDENTIFIER) {
```
Comment (maidul98) [inline cli/packages/cmd/folder.go:128]: I'm wondering if the names of the tokens should be uniform. I think that would make it easy for end users if the token is just set under INFISICAL_TOKEN regardless of auth type. What i'm thinking is the only difference between service token and machine identity tokens is that service tokens have a private key attached to the token but they can both be added as bearer token. What do you think?

File: cli/packages/cmd/secrets.go [hunk 2]
```diff
@@ -160,211 +154,38 @@ var secretsSetCmd = &cobra.Command{
...
+			util.HandleError(err, "Unable to parse secret type")
 		}
 
-		encryptedWorkspaceKey, _ := base64.StdEncoding.DecodeString(workspaceKeyResponse.EncryptedKey)
-		encryptedWorkspaceKeySenderPublicKey, _ := base64.StdEncoding.DecodeString(workspaceKeyResponse.Sender.PublicKey)
-		encryptedWorkspaceKeyNonce, _ := base64.StdEncoding.DecodeString(workspaceKeyResponse.Nonce)
-		currentUsersPrivateKey, _ := base64.StdEncoding.DecodeString(loggedInUserDetails.UserCredentials.PrivateKey)
+		var secretOperations []models.SecretSetOperation
+		if token != nil && (token.Type == util.SERVICE_TOKEN_IDENTIFIER || token.Type == util.UNIVERSAL_AUTH_TOKEN_IDENTIFIER) {
+			secretOperations, err = util.SetRawSecrets(args, secretType, environmentName, secretsPath, projectId, token)
```
Comment (maidul98) [inline cli/packages/cmd/secrets.go:179]: Could it be that the person using service tokens haven't yet upgraded their project?

File: cli/packages/cmd/secrets.go [hunk 3]
```diff
@@ -804,25 +646,17 @@ func init() {
 
 	secretsCmd.Flags().Bool("secret-overriding", true, "Prioritizes personal secrets, if any, with the same name over shared secrets")
 	secretsCmd.AddCommand(secretsSetCmd)
+	secretsSetCmd.Flags().String("token", "", "Fetch secrets using the Infisical Token")
+	secretsSetCmd.Flags().String("projectId", "", "manually set the projectId to fetch folders from for machine identity")
```
Comment (maidul98) [inline cli/packages/cmd/secrets.go:650]: I think you meant `when using machine identity based auth`?

**Review by maidul98 — COMMENTED** [review]
Just left some small comments

File: cli/packages/cmd/folder.go (hunk 1 above)
Comment (DanielHougaard) [reply cli/packages/cmd/folder.go:128]: This is actually handled in the `GetInfisicalToken()` method. It checks the `INFISICAL_TOKEN` env variable

File: cli/packages/cmd/secrets.go (hunk 2 above)
Comment (DanielHougaard) [reply cli/packages/cmd/secrets.go:179]: In the `secrets set` command, I decided to assume that the user attempting to use their service token for setting secrets, has already upgraded their project. I opted for this to reduce complexity to some degree

**Review by maidul98 — APPROVED** [review]

**Pull request merged**